File: shared/src/types/transaction.ts
export interface Transaction {
  id: string;
  user: string;
  amount: number;
  description: string;
  merchant?: string;
  category: string;
  type: 'payment' | 'transfer' | 'airtime' | 'transport' | 'allowance';
  status: 'pending' | 'completed' | 'failed';
  payment_method: 'qr' | 'nfc' | 'card';
  created_at: string;
  updated_at: string;
}

export interface QRCode {
  qr_code: string;
  expires_at: string;
  merchant_id?: string;
}

export interface QRPaymentRequest {
  qr_code: string;
  amount: number;
}

export interface NFCToken {
  token: string;
  expires_at: string;
}

export interface NFCPaymentRequest {
  token: string;
  amount: number;
  merchant_id: string;
}

export interface AirtimePurchaseRequest {
  provider: string;
  phone_number: string;
  amount: number;
}

export interface AirtimeProvider {
  id: string;
  name: string;
  logo_url: string;
}

export interface TransportPurchaseRequest {
  route_id: string;
  amount: number;
}

export interface TransportRoute {
  id: string;
  name: string;
  origin: string;
  destination: string;
  price: number;
}

export interface Merchant {
  id: string;
  name: string;
  category: string;
  logo_url?: string;
  is_verified: boolean;
  created_at: string;
}

File: shared/dart/lib/transaction.dart
class Transaction {
  final String id;
  final String child;
  final double amount;
  final String description;
  final String merchant;
  final String category;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;

  Transaction({
    required this.id,
    required this.child,
    required this.amount,
    required this.description,
    required this.merchant,
    required this.category,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['id'] as String,
      child: json['child'] as String,
      amount: (json['amount'] as num).toDouble(),
      description: json['description'] as String,
      merchant: json['merchant'] as String,
      category: json['category'] as String,
      status: json['status'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'child': child,
      'amount': amount,
      'description': description,
      'merchant': merchant,
      'category': category,
      'status': status,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}
